tests/Http/Offer/UpdateBookingAvailabilityRequestBodyParserTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Offer;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblemException;
+use CultuurNet\UDB3\Http\Request\Psr7RequestBuilder;
+use PHPUnit\Framework\TestCase;
+
+final class UpdateBookingAvailabilityRequestBodyParserTest extends TestCase
+{
+    /**
+     * @var UpdateBookingAvailabilityRequestBodyParser
+     */
+    private $updateBookingAvailabilityRequestBodyParser;
+
+    /**
+     * @var Psr7RequestBuilder
+     */
+    private $requestBuilder;
+
+    protected function setUp(): void
+    {
+        $this->updateBookingAvailabilityRequestBodyParser = new UpdateBookingAvailabilityRequestBodyParser();
+        $this->requestBuilder = new Psr7RequestBuilder();
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_valid_data(): void
+    {
+        $given = $this->requestBuilder->withBodyFromString('{"type":"Available"}')->build('PUT');
+        $expected = [
+            'type' => 'Available',
+        ];
+
+        $actual = $this->updateBookingAvailabilityRequestBodyParser->parse($given);
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @test
+     */
+    public function it_fails_on_empty_body(): void
+    {
+        $given = $this->requestBuilder->withBodyFromString('')->build('PUT');
+        $this->expectException(ApiProblemException::class);
```

Comment:
I wonder if we should also try to validate the `type` of the `ApiProblem` here that should be `https://api.publiq.be/probs/body/missing`

Reply:
I also considered it but then we have to do a `try/catch` and use `$this->fail()` inside the `try` after the line with the `->parse()`. It's doable but a bit dirty. But indeed a better test. What do you think?

Reply:
I guess it can be useful to already verify the type because in the future we will have more types and it's important that they match. I checked the PHPUnit docs but it seems there is no method to get hold of the thrown exception itself. So an extra try/catch seems to be required.

Reply:
I'll see if I can find a more elegant solution and if not I'll do it with a try/catch. Maybe in a reusable trait then so we can use it in other tests to check the problem type.

Reply:
I notice some people have been asking for an `assertException` which has the exception as an input parameter. But that's not part of PHPUnit https://github.com/sebastianbergmann/phpunit/issues/3071

Reply:
Nice approach with the callable inside the trait!